src/library/scala/collection/immutable/Vector.scala

```diff
@@ -799,98 +725,81 @@ private[immutable] trait VectorPointer[T] {
       }
     }
 
-
     // requires structure is at pos oldIndex = xor ^ index
     private[immutable] final def getElem(index: Int, xor: Int): T = {
-      if (xor < (1 << 5)) { // level = 0
+      if        (xor < (1 <<  5)) { // level = 0
         display0(index & 31).asInstanceOf[T]
-      } else
-      if (xor < (1 << 10)) { // level = 1
-        display1((index >> 5) & 31).asInstanceOf[Array[AnyRef]](index & 31).asInstanceOf[T]
-      } else
-      if (xor < (1 << 15)) { // level = 2
-        display2((index >> 10) & 31).asInstanceOf[Array[AnyRef]]((index >> 5) & 31).asInstanceOf[Array[AnyRef]](index & 31).asInstanceOf[T]
-      } else
-      if (xor < (1 << 20)) { // level = 3
-        display3((index >> 15) & 31).asInstanceOf[Array[AnyRef]]((index >> 10) & 31).asInstanceOf[Array[AnyRef]]((index >> 5) & 31).asInstanceOf[Array[AnyRef]](index & 31).asInstanceOf[T]
-      } else
-      if (xor < (1 << 25)) { // level = 4
-        display4((index >> 20) & 31).asInstanceOf[Array[AnyRef]]((index >> 15) & 31).asInstanceOf[Array[AnyRef]]((index >> 10) & 31).asInstanceOf[Array[AnyRef]]((index >> 5) & 31).asInstanceOf[Array[AnyRef]](index & 31).asInstanceOf[T]
-      } else
-      if (xor < (1 << 30)) { // level = 5
-        display5((index >> 25) & 31).asInstanceOf[Array[AnyRef]]((index >> 20) & 31).asInstanceOf[Array[AnyRef]]((index >> 15) & 31).asInstanceOf[Array[AnyRef]]((index >> 10) & 31).asInstanceOf[Array[AnyRef]]((index >> 5) & 31).asInstanceOf[Array[AnyRef]](index & 31).asInstanceOf[T]
-      } else { // level = 6
+      } else if (xor < (1 << 10)) { // level = 1
+        display1((index >>>  5) & 31).asInstanceOf[Array[AnyRef]](index & 31).asInstanceOf[T]
+      } else if (xor < (1 << 15)) { // level = 2
+        display2((index >>> 10) & 31).asInstanceOf[Array[AnyRef]]((index >>>  5) & 31).asInstanceOf[Array[AnyRef]](index & 31).asInstanceOf[T]
+      } else if (xor < (1 << 20)) { // level = 3
+        display3((index >>> 15) & 31).asInstanceOf[Array[AnyRef]]((index >>> 10) & 31).asInstanceOf[Array[AnyRef]]((index >>>  5) & 31).asInstanceOf[Array[AnyRef]](index & 31).asInstanceOf[T]
+      } else if (xor < (1 << 25)) { // level = 4
+        display4((index >>> 20) & 31).asInstanceOf[Array[AnyRef]]((index >>> 15) & 31).asInstanceOf[Array[AnyRef]]((index >>> 10) & 31).asInstanceOf[Array[AnyRef]]((index >>>  5) & 31).asInstanceOf[Array[AnyRef]](index & 31).asInstanceOf[T]
+      } else if (xor < (1 << 30)) { // level = 5
+        display5((index >>> 25) & 31).asInstanceOf[Array[AnyRef]]((index >>> 20) & 31).asInstanceOf[Array[AnyRef]]((index >>> 15) & 31).asInstanceOf[Array[AnyRef]]((index >>> 10) & 31).asInstanceOf[Array[AnyRef]]((index >>> 5) & 31).asInstanceOf[Array[AnyRef]](index & 31).asInstanceOf[T]
```

Comment:
This is an awful mess even with the improvements unless you have a very wide screen.  If you're up for it maybe you can rewrite it in the style
```
display5((index >>> 25) & 31).asInstanceOf[Array[AnyRef]].
   apply((index >>> 20 & 31)).asInstanceOf[Array[AnyRef]].
   apply((index >>> 15) & 31) . . .
```
The alternative--lining up the matching part of the displays--pushes the earlier ones so far right that they disappear, so that's probably not a good idea.